src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlGlobalStaticResourcesMap.cs

```diff
@@ -29,6 +32,108 @@ internal void Add(string staticResourceKey, string ns, ResourcePrecedence preced
 			list.Add(new StaticResourceDefinition(staticResourceKey, ns, precedence));
 		}
 
+		/// <summary>
+		/// Gets the names of all GlobalStaticResources properties associated with a top-level ResourceDictionary.
+		/// </summary>
+		/// <param name="initialFiles">File names of the ResourceDictionaries whose properties should be returned first.</param>
+		/// <remarks>This is used when building Uno.UI itself to create a master dictionary of system resources.</remarks>
+		internal IEnumerable<string> GetAllDictionaryProperties(string[] initialFiles)
+		{
+			var initialProperties = initialFiles.Select(f =>
+					_rdMap.First(kvp =>
+						kvp.Key.EndsWith(f, StringComparison.InvariantCultureIgnoreCase)
+					)
+				)
+				.ToArray();
+
+			return initialProperties.Concat(
+					_rdMap.Except(initialProperties)
+				)
+				.Select(kvp => ConvertIdToResourceDictionaryProperty(kvp.Value.UniqueID));
+		}
+
+		/// <summary>
+		/// Gets the name of a GlobalStaticResources property associated with a ResourceDictionary.Source designation. Throws an exception if none is found.
+		/// </summary>
+		/// <param name="originDictionary">The file containing the XAML</param>
+		/// <param name="source">The ResourceDictionary.Source value</param>
+		internal string FindTargetPropertyForMergedDictionarySource(XamlFileDefinition originDictionary, string source)
+		{
+			var targetSource = GetSourceLink(originDictionary);
+			var absoluteSource = ResolveAbsoluteSource(targetSource, source);
+
+			if (_rdMap.TryGetValue(absoluteSource, out var xamlFileDefinition))
+			{
+				// TODO: check if it actually corresponds to a ResourceDictionary
+				return ConvertIdToResourceDictionaryProperty(xamlFileDefinition.UniqueID);
+			}
+
+			return null;
+		}
+
+		/// <summary>
+		/// Get the appropriate source link for a given XAML file.
+		/// </summary>
+		internal string GetSourceLink(XamlFileDefinition xamlFileDefinition)
+		{
+			return _rdMap.FirstOrDefault(kvp => kvp.Value == xamlFileDefinition).Key; //TODO: this is O(n), is it an actual perf issue?
+		}
+
+		/// <summary>
+		/// Convert relative source path to absolute path.
+		/// </summary>
+		private string ResolveAbsoluteSource(string origin, string relativeTargetPath)
+		{
+			if (relativeTargetPath.StartsWith(AppXIdentifier))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Change this call to 'relativeTargetPath.StartsWith' to an overload that accepts a 'StringComparison' as a parameter.](https://app.codacy.com/gh/unoplatform/uno/pullRequest?prid=4826539)